lib/shopify_app/controller_concerns/webhook_verification.rb

```diff
@@ -15,12 +15,18 @@ def verify_request
     end
 
     def hmac_valid?(data)
-      secret = ShopifyApp.configuration.secret
-      digest = OpenSSL::Digest.new('sha256')
-      ActiveSupport::SecurityUtils.secure_compare(
-        shopify_hmac,
-        Base64.encode64(OpenSSL::HMAC.digest(digest, secret, data)).strip
-      )
+      secrets = [ShopifyApp.configuration.secret, ShopifyApp.configuration.old_secret].map(&:to_s)
```

Comment:
I don't think you want `to_s` here. Are you guarding against `nil`? Because if so you basically want:

```ruby
secrets = [ShopifyApp.configuration.secret, ShopifyApp.configuration.old_secret].reject(&:blank)
```

Reply:
That was exactly why. Thanks!